### PR TITLE
Generate parents for subjects created through uploads

### LINKED ISSUE #1444: Parents sub-doc does not get populated in label/reaper uploads
Likely affects SDK gear rules permission checks.

Added [checks in integration tests](https://github.com/flywheel-io/core/commit/f24e35a3652930950157129e176801ba780ccf02) on the `no-parents` branch to illustrate the problem.

We will need to fix this, and add an upgrade script to correct the affected files.

### PR DESCRIPTION
**Pull request by hkethi002**
Fixes #1444 
### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

**Comment by ehlertjd** [comment]
Still probably need a db-upgrade to fix existing reaper uploads.

**Comment by nagem** [comment]
It looks like only those on EM5 (4.2.x) should be affected by this bug, correct?

File: bin/database.py [hunk 1]
```diff
@@ -1931,6 +1931,41 @@ def upgrade_to_56():
...
+    CHILD_MAP ={
+        "subjects": "sessions",
+        "sessions": "acquisitions"
+    }
+
+    cont_type = containerutil.singularize(cont_name)
+
+    # Handle subject without parents
+    if cont_type == 'subject':
+        subject_group = config.db.projects.find_one({'_id': cont['project']})
```
Comment (nagem) [inline bin/database.py]: A little confusingly named, but since it's a db update that's a low priority.

File: bin/database.py [hunk 2]
```diff
@@ -1931,6 +1931,41 @@ def upgrade_to_56():
...
+            group_id = subject_group['group']
+        parents = {'project': cont['project'], 'group': group_id}
+        config.db.subjects.update({'_id': cont['_id']}, {'$set': {'parents': parents}})
+    else:
+        parents = cont.get('parents', {})
+
+    parents[cont_type] = cont['_id']
+    config.db['analyses'].update_many({'parent.id': cont['_id']}, {'$set': {'parents': parents}})
+
+    if cont_name != 'acquisitions':
```
Comment (nagem) [inline bin/database.py]: Why would cont_name ever be 'acquisitions'? It looks like upgrade_to_57 is only called on subjects.

Comment (nagem) [inline bin/database.py]: My previous comment was incorrect, I see it is recursively updating all children here.

**Pull request merged**